js/pie_utils.js

```diff
@@ -1,101 +1,47 @@
 // functions related to the pie charts
 
-var pie = d3.pie()
-  .sort(null)
-  .value(function(d) { return d.value; });
-
 var arcpath = d3.arc()
-  .innerRadius(0);
+  .innerRadius(0)
+  .outerRadius(1);
 
-function addPieCharts() {
+function addPieSlices() {
   
-  //relies on map and pieFormData as a global variable
-  
-  // each pie is a group of pie slices. here add the pie groups
-  var pies = map.selectAll('.pie')
-    .data(pieFormData);
-  var piesMerged = pies
-    .enter()
-    .append('g')
-      .classed("pie", true)
-      .attr("transform", function(d) {
-        return "translate(" + d.coordinates.x + "," + d.coordinates.y + ")";
-      })
-      .merge(pies);
+  //relies on map as a global variable
   
   // add pie slices to each pie group
-  piesMerged.selectAll('.pieslice')  
+  map.selectAll('g.pie').selectAll('.slice')  
     .data(function(d) {
       return d.sliceGeomData;
     })
     .enter()
     .append("path")
-      .classed("pieslice", true);
+      .classed("slice", true)
+      .style("fill", function(d) { 
+        return categoryToColor(d.data.category); 
+      })
+      .attr("d", arcpath);
   
   piesBaked = true;
   
   // these newly added pie charts won't work until updatePieCharts is called, but
   // since this function always gets called from updatePieCharts, that should be fine
 }
 
-function updatePieCharts() {
+function updatePieSlices(category, delay, duration) {
   
   // add the pies if needed
   if (!piesBaked) {
-    addPieCharts();
+    addPieSlices();
   }
   
-  map.selectAll('.pie').selectAll('.pieslice')
-    //.transition().duration(0)
-    .attr("d", arcpath.outerRadius(function(d) {
-      return scaleCircles(d.data.total);
-    }))
-    .attr("fill", function(d) { 
-      return categoryToColor(d.data.category); 
-    });
-  
+  if(category === 'piechart') {
+    d3.selectAll(".slice")
+      .transition().delay(delay).duration(duration)
+      .style("display", null);
```

Comment:
I think you might want `block` here?

Reply:
can you tell me why? i don't know much about the "display" options, but i think when i first create the item there is no `display` info inline at all, so i just set it back to not being in there by assigning `null`

Reply:
cool, thanks for the link!